File: Contracts/src/main/java/org/cubexmc/contract/gui/Signature.java
package org.cubexmc.contract.gui;

import java.util.Locale;

/**
 * Signature token validation for the anvil confirmation layer. A signature is
 * accepted when the player types their own name or one of the agreement words.
 * Kept as a small pure function so the rule can be unit tested.
 */
public final class Signature {
    private Signature() {
    }

    public static boolean matches(String playerName, String input) {
        if (playerName == null || input == null) {
            return false;
        }
        String token = input.trim();
        if (token.isEmpty()) {
            return false;
        }
        if (token.equalsIgnoreCase(playerName.trim())) {
            return true;
        }
        String lower = token.toLowerCase(Locale.ROOT);
        return token.equals("同意") || lower.equals("agree") || lower.equals("confirm");
    }
}

File: Contracts/src/main/java/org/cubexmc/contract/gui/Signature.kt
package org.cubexmc.contract.gui

import java.util.Locale

object Signature {
    @JvmStatic
    fun matches(playerName: String?, input: String?): Boolean {
        if (playerName == null || input == null) {
            return false
        }
        val token = input.trim()
        if (token.isEmpty()) {
            return false
        }
        if (token.equals(playerName.trim(), ignoreCase = true)) {
            return true
        }
        val lower = token.lowercase(Locale.ROOT)
        return token == "同意" || lower == "agree" || lower == "confirm"
    }
}
